website/docs/cloud-docs/api-docs/stacks.mdx

```diff
@@ -0,0 +1,493 @@
+| `data.attributes.vcs-repo.oauth-token-id`      | string  |                  | The VCS Connection (OAuth Connection + Token) to use. This ID can be obtained from the [oauth-tokens](/terraform/cloud-docs/api-docs/oauth-tokens) endpoint.                                                                                  |
+| `data.relationships.project.data.id`           | string  |                  | The ID of the project to create the stack in. You must have permission to manage the project, either by organization-level permissions or team admin access to a specific project.                                                            |
+
+### Sample Payload
+```json
+{
+  "data": {
+    "attributes": {
+      "vcs-repo": {
+        "identifier": "hashicorp/pet-nulls-stack-local",
+        "display-identifier": "hashicorp/pet-nulls-stack-local",
+        "oauth-token-id": "ot-hmAyP66qk2AMVdbJ",
+        "branch": "",
+        "service-provider": "github",
+        "repository-http-url":"https://github.com/hashicorp/pet-nulls-stack-local",
+        "tags-regex": null,
+        "trigger-disabled": false
+      },
+      "name": "pet-nulls-stack-local",
+      "description": "testing patch",
+      "locked"  : false,
+      "lock-reason": "",
+      "speculative-enabled": false
+    }
+  }
+}
+```
+### Sample Request
+
+```shell
+$ curl\
+  --header "Authorization: Bearer $TOKEN" \
+  --header "Content-Type: application/vnd.api+json" \
+  --request PATCH \
+  https://app.terraform.io/api/v2/stacks/st-EJyw2MktbRjdsg96
+```
+
+### Sample Response
+```json
+{
+  "data": {
+    "id": "st-EJyw2MktbRjdsg96",
+    "type": "stacks",
+    "attributes": {
+      "name": "pet-nulls-stack-local",
+      "description": "testing patch",
+      "locked"  : false,
+      "lock-reason": "",
+      "linked-stack-connections": {
+        "upstream-count": 0,
+        "downstream-count": 0,
+        "inputs-count": 0,
+        "outputs-count": 0
+      },
+      "speculative-enabled": false,
+      "vcs-repo": {
+        "branch": "",
+        "tags-regex": null,
+        "identifier": "hashicorp/pet-nulls-stack-local",
+        "display-identifier": "hashicorp/pet-nulls-stack-local",
+        "oauth-token-id": "ot-t3remoFJPeuLk1Mz",
+        "repository-http-url": "https://github.com/hashicorp/pet-nulls-stack-local",
+        "service-provider": "github",
+        "trigger-disabled": false
+      },
+      "created-at": "2025-05-09T16:40:31.255Z",
+      "updated-at": "2025-05-09T20:24:47.324Z"
+    },
+    "relationships": {
+      "project": {
+        "data": {
+          "id": "prj-18PG93QyqCMhLhWT",
+          "type": "projects"
+        }
+      }
+    },
+    "links": {
+      "self": "/api/v2/stacks/st-EJyw2MktbRjdsg96",
+      "stack-configurations": "/api/v2/stacks/st-EJyw2MktbRjdsg96/stack-configurations",
+      "stack-deployments": "/api/v2/stacks/st-EJyw2MktbRjdsg96/stack-deployments"
+    }
+  }
+}
+```
+
+## Delete a Stack
+
+`DELETE /stacks/:policy_id`
```

Comment:
typo: `POST /stacks/:stack_id`

Reply:
@aaabdelgany This is updated as DELETE in the openAPI swagger docs which I am using as the source of truth for these API docs, has this changed? What should I be using as the source of truth? : https://editor.swagger.io/?url=https%3A%2F%2Fgist.githubusercontent.com%2Fjondavidjohn%2F53831292fae6bcb14285839bda45c683%2Fraw%2Fswagger.json 

Also this document attached to the ticket says the same thing: https://docs.google.com/document/d/17CsTSIA0QP7QEQjM1EqWNQ9W9gsHv4ZBAh3H4wf5io8/edit?tab=t.0

Reply:
It probably should be a DELETE, I didn't realize that the openapi docs also specified that. I'll make a ticket to throw in the backlog and make sure this eventually gets changed to a DELETE.

Reply:
@aaabdelgany Awesome! 🙇‍♀️ Feel free to approve if no more changes required